packages/esm-appointments-app/src/workload/monthly-view-workload/monthly-view.component.tsx

```diff
@@ -41,7 +42,7 @@ const MonthlyCalendarView: React.FC<MonthlyCalendarViewProps> = ({
           <div className={styles.monthlyCalendar}>
             {monthDays(dayjs(monthViewDate)).map((dateTime, i) => (
               <div
-                onClick={() => handleClick(dayjs(dateTime).format('YYYY-MM-DD'))}
+                onClick={() => handleClick(dayjs(dateTime).toString())}
```

Comment:
Thanks for tracking this down, and generally looks good, @usamaidrsk , but I'm wondering, would it still work (and be better) if we passed in dayjs(datetime).toDate() to handleClick, and then changed handleClick on line 24 to just:

```
  const handleClick = (date: Date) => {
    if (onDateClick) {
      onDateClick(date);
    }
  };
```

It seems like the whole problem occurs because we try to go from dayjs object -> string -> Date object, while in the above we can just go from dayjs object -> Date.

Thoughts?

Reply:
Thanks @mogoodrich , great suggestion. I will update to that.